src/main/java/com/google/api/generator/gapic/model/Message.java

```diff
@@ -80,6 +89,32 @@ public boolean hasResource() {
     return resource() != null;
   }
 
+  public void validateField(String fieldName, Map<String, Message> messageTypes) {
+    String[] subFields = fieldName.split("\\.");
+    Message nestedMessage = this;
+    for (int i = 0; i < subFields.length; i++) {
+      String subFieldName = subFields[i];
+      if (i < subFields.length - 1) {
+        Field field = nestedMessage.fieldMap().get(subFieldName);
+        nestedMessage = messageTypes.get(field.type().reference().fullName());
+        Preconditions.checkNotNull(
+            nestedMessage,
+            String.format(
+                MESSAGE_NOT_FOUND_ERROR_MESSAGE,
+                field.name(),
+                field.type().reference().simpleName()));
+      } else {
+        Preconditions.checkState(
```

Comment:
Why do we check this only for the last `subFieldName`? Should we do this for all?